File: app/src/main/java/com/sayler/gina/dagger/component/DataComponent.java
package com.sayler.gina.dagger.component;

import com.sayler.gina.activity.DayActivity;
import com.sayler.gina.activity.DayEditActivity;
import com.sayler.gina.activity.MainActivity;
import com.sayler.gina.dagger.module.ServicesModule;
import com.sayler.ormliteimplementation.DataModuleOrmLite;
import dagger.Component;

import javax.inject.Singleton;

@Singleton
@Component(
    modules = {
        DataModuleOrmLite.class,
        ServicesModule.class
    },
    dependencies = {
        ApplicationComponent.class
    }
)
public interface DataComponent {

  /* ---- ACTIVITY ---- */
  void inject(MainActivity mainActivity);

  void inject(DayActivity mainActivity);

  void inject(DayEditActivity mainActivity);

  /* ---- FRAGMENT ---- */


  /* ---- PRESENTER ---- */
}

File: app/src/main/java/com/sayler/gina/dagger/component/DataComponent.kt
package com.sayler.gina.dagger.component

import com.sayler.gina.GinaApplication
import com.sayler.gina.activity.DayActivity
import com.sayler.gina.activity.EditDayActivity
import com.sayler.gina.activity.MainActivity
import com.sayler.gina.dagger.module.ManagersModule
import com.sayler.ormliteimplementation.DataModuleOrmLite
import dagger.Component

import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(DataModuleOrmLite::class, ManagersModule::class), dependencies = arrayOf(ApplicationComponent::class))
interface DataComponent {

    /* ---- ACTIVITY ---- */
    fun inject(mainActivity: MainActivity)

    fun inject(mainActivity: DayActivity)

    fun inject(mainDayActivity: EditDayActivity)

    fun inject(ginaApplication: GinaApplication)

    /* ---- FRAGMENT ---- */


    /* ---- PRESENTER ---- */
}